### ISSUE TITLE
Unicode functions are not TypedArray safe.

### ISSUE THREAD

**Issue by curiousdannii** [erkyrath/glkote#27]
The truncation tests in unicasetest.inf fail if glk_buffer_canon_normalize_uni is passed a Uint32Array. Possibly the other functions would fail under the right circumstances, so I'm going to pass normal arrays just to be safe.

**Comment by erkyrath** [comment]
Right, all of that code was written before I had any experience of typed arrays.

One day there will be a big conversion effort.
